org/postgresql/jdbc2/AbstractJdbc2ResultSet.java

```diff
@@ -488,7 +488,7 @@ public int getConcurrency() throws SQLException
                 return connection.getTimestampUtils().toDateBin(tz, this_row[col]);
             } else if (oid == Oid.TIMESTAMP || oid == Oid.TIMESTAMPTZ) {
                 // JDBC spec says getDate of Timestamp must be supported
-                return connection.getTimestampUtils().convertToDate(getTimestamp(i, cal), tz);
+                return connection.getTimestampUtils().convertToDate(getTimestamp(i, Calendar.getInstance(TimeZone.getTimeZone("UTC"))), tz);
```

Comment:
Updated the patch, thanks for looking over my shoulder

Dave Cramer

On 29 September 2015 at 12:23, Dave Cramer <davecramer@gmail.com> wrote:

> Hmmm I thought I did that....
>
> Dave Cramer
>
> On 29 September 2015 at 12:22, Vladimir Sitnikov <notifications@github.com
> > wrote:
>
>> In org/postgresql/jdbc2/AbstractJdbc2ResultSet.java
>> <https://github.com/pgjdbc/pgjdbc/pull/376#discussion_r40693683>:
>>
>> > @@ -488,7 +488,7 @@ public int getConcurrency() throws SQLException
>> >                  return connection.getTimestampUtils().toDateBin(tz, this_row[col]);
>> >              } else if (oid == Oid.TIMESTAMP || oid == Oid.TIMESTAMPTZ) {
>> >                  // JDBC spec says getDate of Timestamp must be supported
>> > -                return connection.getTimestampUtils().convertToDate(getTimestamp(i, cal), tz);
>> > +                return connection.getTimestampUtils().convertToDate(getTimestamp(i, Calendar.getInstance(TimeZone.getTimeZone("UTC"))), tz);
>>
>> I do not like calendar.getinstance.
>> If "UTC" calendar is required, it should be cached inside TimestampUtils.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/pgjdbc/pgjdbc/pull/376/files#r40693683>.
>>
>
>